### PR TITLE
fixed the issue where the original name of a node would be truncated when the name was too long.

### LINKED ISSUE #647: The name of folders can get truncated upon renaming them
Under certain conditions, the original name of a folder might be lost if we try to rename it.

**Steps to reproduce:**

1. Make the folders panel as narrow as possible
2. Create a folder with a long name, e.g.: `This is a really long name for a folder`
3. Try renaming that folder

**What will happen:**

The original name of the folder will be be lost. In this example, it will be renamed to `This is a really…` and there's no way to prevent it.

**What should have happened:**

The original name of the folder should be kept.

**Video:**

https://github.com/nuttyartist/notes/assets/626206/724defb6-2b50-4f0c-96d0-9b62adb2dd98

### PR DESCRIPTION
**Pull request by 1250890838**
fixed #647

### CODE REVIEW

**Comment by nuttyartist** [comment]
@1250890838 Thanks a lot, it works well (:

**Review by nuttyartist — APPROVED** [review]

File: src/elidedlabel.cpp [hunk 1]
```diff
@@ -3,18 +3,11 @@
...
-{
-    defaultType = Qt::ElideRight;
-    eliding = false;
-    original = "";
-}
+ElidedLabel::ElidedLabel(QWidget *parent, Qt::WindowFlags f) : ElidedLabel("", parent, f) { }
 
 ElidedLabel::ElidedLabel(const QString &text, QWidget *parent, Qt::WindowFlags f)
-    : QLabel(text, parent, f)
+    : QLabel(text, parent, f), defaultType(Qt::ElideRight), eliding(false), original("")
```
Comment (zjeffer) [inline src/elidedlabel.cpp:9]: Can you fix the warning?

```
In file included from /src/src/elidedlabel.cpp:1:
/src/src/elidedlabel.h: In constructor 'ElidedLabel::ElidedLabel(const QString&, QWidget*, Qt::WindowFlags)':
/src/src/elidedlabel.h:26:10: warning: 'ElidedLabel::eliding' will be initialized after [-Wreorder]
   26 |     bool eliding;
      |          ^~~~~~~
/src/src/elidedlabel.h:24:13: warning:   'QString ElidedLabel::original' [-Wreorder]
   24 |     QString original;
      |             ^~~~~~~~
/src/src/elidedlabel.cpp:8:1: warning:   when initialized here [-Wreorder]
    8 | ElidedLabel::ElidedLabel(const QString &text, QWidget *parent, Qt::WindowFlags f)
      | ^~~~~~~~~~~
```

**Review by zjeffer — CHANGES_REQUESTED** [review]

**Comment by 1250890838** [comment]
ok,i have fixed it.

**Review by guihkx — APPROVED** [review]
Also confirming the issue is fixed. Thanks!

**Pull request merged**